src/os/bluestore/KernelDevice.h

```diff
@@ -26,7 +26,7 @@
 #include "BlockDevice.h"
 
 #ifndef RW_IO_MAX
-#define RW_IO_MAX 0x7FFFF000
+#define RW_IO_MAX 0x7FFF0000
```

Comment:
the upper bound of total size of iov can be transferred in a `pwritev()` call is checked and capped by `rw_copy_check_uvector()` in Linux. the limit is `MAX_RW_COUNT`.  see https://github.com/torvalds/linux/blob/451577f3e3a9bf1861218641dbbf98e214e77851/fs/read_write.c#L828

but unfortunately, glibc does not offer a macro/function to expose this setting. so for maximizing the performance, we need to calculate/define this by ourselves.

and `MAX_RW_COUNT` is defined here:

https://github.com/torvalds/linux/blob/6e6d05360b80f196ed07061327f03346b204abea/include/linux/fs.h#L2305

just to be 100% sure, https://github.com/torvalds/linux/blob/30d1d92a888d03681b927c76a35181b4eed7071f/tools/virtio/linux/kernel.h#L23

so we need to use `CEPH_PAGE_MASK` for defining this variable instead of hardwiring it to `0x7FFF0000`. something like:
```c++
// see MAX_RW_COUNT in linux/include/linux/fs.h
#define RW_IO_MAX (INT_MAX & CEPH_PAGE_MASK)
```

Reply:
also i don't think we need to guard this macro with `#ifndef` as it's defined nowhere else.

Reply:
ah, thanks! that makes way more sense.